Sources/MapboxCoreNavigation/EHorizon/RoadGraph.swift

```diff
@@ -32,6 +32,35 @@ public final class RoadGraph {
         }
         return LineString(locations.map { $0.coordinate })
     }
+    
+    /**
+     Returns a line string geometry corresponding to the given path.
+     
+     - returns: A line string corresponding to the given path, or `nil` if any of path edges is inaccessible.
+     */
+    public func pathShape(for path: Path) -> LineString? {
+        let nativePath = GraphPath(edges: path.edgeIdentifiers.map { NSNumber(value: $0) },
+                                   percentAlongBegin: path.fractionFromStart,
+                                   percentAlongEnd: path.fractionToEnd,
+                                   length: path.length)
+        guard let locations = native.getPathShape(for: nativePath) else {
+            return nil
+        }
+        return LineString(locations.map { $0.coordinate })
+    }
+    
+    /**
+     Returns a point corresponding to the given position.
+     
+     - returns: A point corresponding to the given position, or `nil` if the edge is inaccessible.
+     */
+    public func position(for position: Position) -> Point? {
```

Comment:
“Position” is what’s being passed in, not what’s returned, so rename to `shape(of:)`. That’s the same signature as the method above, which accomplishes the same effect as what I suggested in https://github.com/mapbox/mapbox-navigation-ios/issues/3035#issuecomment-850520609.

Reply:
Renamed to `shape(of position: Position) -> Point?`. Hope I got your point correctly.